code/game/machinery/kitchen/microwave.dm

```diff
@@ -115,23 +115,24 @@
 			return 1
 	else if(src.dirty==100) // The microwave is all dirty so can't be used!
 		var/obj/item/weapon/reagent_containers/R = O
-		if(istype(R)) // If they're trying to clean it then let them
-			if(R.reagents.amount_cache.len == 1 && R.reagents.has_reagent(CLEANER, 5))
+		var/obj/item/weapon/soap/S = O
+		if((istype(R) && (R.reagents.amount_cache.len == 1 && R.reagents.has_reagent(CLEANER, 5))) || istype(S)) // If they're trying to clean it then let them
+			user.visible_message( \
+				"<span class='notice'>[user] starts to clean the microwave.</span>", \
+				"<span class='notice'>You start to clean the microwave.</span>" \
+			)
+			if (do_after(user, src,20))
+				if(istype(R))
+					R.reagents.remove_reagent(CLEANER,5)
 				user.visible_message( \
-					"<span class='notice'>[user] starts to clean the microwave.</span>", \
-					"<span class='notice'>You start to clean the microwave.</span>" \
+					"<span class='notice'>[user]  has cleaned  the microwave.</span>", \
```

Comment:
too many spaces

Reply:
What do you mean? The relevant section wasn't changed.

Reply:
Oh I see the extra spaces now, should be fixed.